src/alloc.c

```diff
@@ -476,8 +476,6 @@ static jl_lambda_info_t *jl_clone_thunk(jl_lambda_info_t *linfo, jl_tupletype_t
     new_linfo->nargs = linfo->nargs;
     new_linfo->isva = linfo->isva;
     new_linfo->rettype = linfo->rettype;
-    new_linfo->def = NULL;
-    new_linfo->specTypes = types;
```

Comment:
Well, this is a static function, plus `jl_new_lambda_info_uninit` also leaves this field set to NULL. Also, I don't want to see more uses of this function. IMO, the public API of this subsystem is basically method_table_insert and get_specialization.